src/bridge/test-metrics.c

```diff
@@ -719,6 +719,74 @@ test_deprecated_net_all (void)
   g_object_unref (transport);
 }
 
+static void
+test_sched_load (void)
+{
+  MockTransport *transport = mock_transport_new ();
+  CockpitChannel *channel;
+  JsonObject *options = json_obj ("{ 'metrics': [ { 'name': 'sched.loadavg' } ],"
+                                  "  'interval': 1000"
+                                  "}");
+  GBytes *msg;
+  JsonObject *res, *description;
+  JsonArray *metrics, *instances;
+  JsonArray *samples, *first_sample;
+
+  g_signal_connect (transport, "closed", G_CALLBACK (on_transport_closed), NULL);
+
+  channel = g_object_new (cockpit_internal_metrics_get_type (),
+                          "transport", transport,
+                          "id", "1234",
+                          "options", options,
+                          NULL);
+
+  cockpit_channel_prepare (channel);
+
+  /* receive meta information */
+  while ((msg = mock_transport_pop_channel (transport, "1234")) == NULL)
+    g_main_context_iteration (NULL, TRUE);
+  res = cockpit_json_parse_bytes (msg, NULL);
+  g_assert (res != NULL);
+  g_assert (json_object_has_member (res, "metrics"));
+
+  /* metrics should have the form [{"name":"sched.loadavg","instances":["1min", "5min", "15min"]}] */
```

Comment:
Don't we have a `assert_json_equal` kind of function that does the right thing?

Reply:
I can't use `cockpit_assert_json_eq()` here as the order of the strings is somewhat random. It always *seems* to be "5min,15min,1min", but this looks like some internal glib happenstance which I don't want to pin down in tests.

Reply:
Something about this wasn't sitting right with me so I dug into it deeper.

You might be using the samples interface incorrectly.  The (1,5,15) minute values from the load-average should properly be considered as a tuple.  This puts it on the order of something like the `free`, `used`, `cached`, `swap-used`, etc. values for `memory`.  Those are all handled as separate fields like `memory.free`, `memory.used`, `memory.cached`, etc

All uses of the second parameter that I've found are for iteration over things that are truly generic, such as lists of cgroups in the system, lists of mount points in the system, list of network interfaces, etc.  In that case, the order is indeed arbitrary.

I'd change the way you record samples to instead record them like `loadavg.1min`, etc. with the second parameter always NULL.  Let's discuss :)

Reply:
I'd like to keep them consistent with PCP's `kernel.all.load` metric, which uses the exact same three instances. That way, JS code can handle a PCP or internal metric channel in the same way, and often code actually ties both as a fallback. [example](https://github.com/cockpit-project/cockpit/blob/master/pkg/dashboard/list.js#L60)_

Reply:
> I'd like to keep them consistent with PCP's kernel.all.load metric, which uses the exact same three instances. 

Oh, that's bad... I fully agree with @allisonkarlitskaya that the 1 minute load average should be proper metric, and not an instance.